source/server/server_configuration_parser.cpp

```diff
@@ -287,6 +293,41 @@ int ServerConfigurationParser::parse_port_with_key(const std::string& key) const
   return parsed_port;
 }
 
+MonikerStreamProcessor ServerConfigurationParser::parse_moniker_stream_processor() const
+{
+    MonikerStreamProcessor stream_processor;
+
+    auto core_config_it = config_file_.find(kMonikerStreamProcessorKey);
+    if (core_config_it != config_file_.end()) {
+        stream_processor.moniker_sideband_stream_read_write = parse_moniker_stream_processor_with_key(kMonikerSidebandStreamReadWriteKey);
+        stream_processor.moniker_stream_write = parse_moniker_stream_processor_with_key(kMonikerStreamWriteKey);
+        stream_processor.moniker_stream_read = parse_moniker_stream_processor_with_key(kMonikerStreamReadKey);
+        stream_processor.moniker_stream_read_write = parse_moniker_stream_processor_with_key(kMonikerStreamReadWriteKey);
+    }
+    return stream_processor;
+}
+
+int ServerConfigurationParser::parse_moniker_stream_processor_with_key(const std::string& key) const
+{
+    int parsed_core = -1;
```

Comment:
I think just `value` would've been sufficient for this, but I see all other methods have `parsed_<name>` variables. `parsed_cpu` will be better name.

Reply:
Updated